sql/sql_parse.cc

```diff
@@ -6362,6 +6362,73 @@ mysql_execute_command(THD *thd)
   DBUG_RETURN(res || thd->is_error());
 }
 
+struct String_pair
+{
+  String_pair(const char *f, const char *s) : first(f), second(s) {}
+  const char *first;
+  const char *second;
+};
+
+static bool get_table_fields(THD *thd, LEX_STRING db, LEX_STRING table_name,
+                             Dynamic_array<const char *> &fields)
+{
+  TABLE_LIST tl;
+  tl.init_one_table(LEX_STRING_WITH_LEN(db), LEX_STRING_WITH_LEN(table_name),
+                    table_name.str, TL_READ);
+  Open_tables_backup open_tables_backup;
+  thd->reset_n_backup_open_tables_state(&open_tables_backup);
+  bool can_deadlock= thd->mdl_context.has_locks();
+  bool error= open_tables_only_view_structure(thd, &tl, can_deadlock);
+  if (!error)
+  {
+    fields.clear();
+    TABLE_SHARE *s= tl.table->s;
+    for (size_t i= 0; i < s->fields; i++)
+    {
+      // TODO try remove strdup()
+      if (fields.append(thd->strdup(s->field[i]->field_name)))
+        return true;
+    }
+  }
+  close_thread_tables(thd);
+  thd->mdl_context.rollback_to_savepoint(
+      open_tables_backup.mdl_system_tables_svp);
```

Comment:
Fix formatting here and in all other places. Notify when done.

Reply:
Fixed some places. Am I missed something?